File: 2/2000s/27s/2799/CountCompleteSubarraysInAnArray.java
/*
https://leetcode.com/problems/count-complete-subarrays-in-an-array/description/?envType=daily-question&envId=2025-04-24
*/

class Solution {

    public static boolean isAValidSet(int nwStSize, int sizeToCheck){
        return nwStSize==sizeToCheck;
    }

    public int countCompleteSubarrays(int[] nums) {
        int lengthOfNums=nums.length, resultantCompSubarrays=0;
        HashSet<Integer>distinctEleInNumsSt=new HashSet<>();
        for(int num: nums){
            distinctEleInNumsSt.add(num);
        }
        int sizeToCheck=distinctEleInNumsSt.size();
        for(int i=0; i<lengthOfNums; ++i){
            HashSet<Integer>currentSeenEleSt=new HashSet<>();
            for(int j=i; j<lengthOfNums; ++j){
                currentSeenEleSt.add(nums[j]);
                int nwSizeToCheck=currentSeenEleSt.size();
                if(isAValidSet(nwSizeToCheck, sizeToCheck)) resultantCompSubarrays++;
                else if(nwSizeToCheck>sizeToCheck) break;
            }
        }
        return resultantCompSubarrays;
    }
}

File: 2/2000s/27s/2799/CountCompleteSubarraysInAnArray.cpp
/*
https://leetcode.com/problems/count-complete-subarrays-in-an-array/description/?envType=daily-question&envId=2025-04-24
*/

class Solution {
public:

    bool isValid(set<int>&st, int sztochk){
        return st.size()==sztochk;
    }

    int countCompleteSubarrays(vector<int>& nums) {
        int sz=nums.size(), res=0;
        set<int>st;
        for(auto &vals: nums){
            st.insert(vals);
        }
        int szst=st.size();
        for(int i=0; i<sz; ++i){
            set<int>curr;
            for(int j=i; j<sz; ++j){
                curr.insert(nums[j]);
                if(isValid(curr, szst)) res++;
                else if(st.size()>szst) break;
            }
        }
        return res;
    }
};